File: Shop.Domain/Entities/Order.cs
using Shop.Domain.Enum;
using System.ComponentModel.DataAnnotations.Schema;

namespace Shop.Domain.Entities
{
    public class Order : BaseEntity
    {
        // address & user information
        public string? Address { get; set; }
        public string? Phone { get; set; }
        public required string UserId { get; set; }
        [ForeignKey("UserId")]
        public AppUser? AppUser { get; set; }
        public string? FullName { get; set; }

        // discount information
        public Discount? Discount { get; set; }
        public int? DiscountId { get; set; }
        public decimal? DiscountPrice { get; set; }

        // shipping method information
        public decimal ShippingFee { get; set; }
        public int ShippingMethodId { get; set; }
        [ForeignKey("ShippingMethodId")]
        public ShippingMethod? ShippingMethod { get; set; }

        // order information
        public decimal SubTotal { get; set; } // gía tất cả sản phẩm
        public OrderStatus Status { get; set; } = OrderStatus.Unconfirmed;
        public DateTime Created { get; set; } = DateTime.Now;
        public DateTime Updated { get; set; }
        public List<OrderItems> OrderItems { get; set; } = new List<OrderItems>();
        public string? Description { get; set; }

        //payment method
        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.Offline;
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Pending;

        public string? StripeSessionId { get; set; }

        public decimal GetTotal()
        {
            // giá tất cả sản phẩm + phí ship - giả giảm
            return SubTotal + ShippingFee - DiscountPrice ?? 0;
        }
    }
}

File: Client/src/app/_models/order.ts
import { CartItem } from './cart';
export interface Order {
  id?: number;
  subTotal: number;
  address: string;
  fullName: string;
  phone: string;
  shippingFee: string;
  shippingMethodId: number;
  shippingMethodDescription: string;
  shippingName: string;
  created: string;
  updated: string;
  userId: string;
  discountId?: number;
  discountPrice?: number;
  orderItems: OrderItem[];
  paymentMethod: string;
  description: string;
  strippeSessionId: string;
  status: string;
  paymentStatus: string;
}

export interface OrderAdd {
  // address & user information
  address: string;
  phone: string;
  fullName: string;
  // shipping method information
  shippingFee: number;
  shippingMethodId: number;
  // discount information
  discountId?: number;
  discountPrice?: number;
  // order information
  description: string;
  subTotal: number;
  // list item
  items: CartItem[];
  paymentMethod: string;
}

export interface OrderItem {
  id: number;
  productId: number;
  quantity: number;
  price: number;
  productName: number;
  sizeName: number;
  colorName: number;
  orderId: number;
  productImage: string;
}

export class OrderParams {
  pageNumber: number = 1;
  pageSize: number = 5;
  search: string = '';
  orderBy: string = 'id_desc';
  selectedPaymentStatus: string = '';
  selectedStatus: string = '';
  userId: string = '';
  startDate: string = '';
  endDate: string = '';
}
